.github/workflows/ci-backwards-compatibility.yml

```diff
@@ -43,11 +43,18 @@ jobs:
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
         continue-on-error: true
 
+      # caching ~/.cargo to avoid downloading all (especially git) deps
+      - name: Cache ~/.cargo
+        uses: actions/cache@v4
+        with:
+          path: ~/.cargo
+          key: ${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
+
       - name: Backwards-compatibility tests
         run: |
           # the default tmp dir is too long (/home/ubuntu/actions-runner/_work/_temp/)
           # we need to use `nix develop -c` to be able to use `nix build` inside of backwards-compatibility-test
-          env TMPDIR=/tmp nix develop -c ./scripts/tests/test-ci-all.sh v0.2.1
+          env TMPDIR=/tmp CARGO_PROFILE=ci nix develop -c ./scripts/tests/test-ci-all.sh v0.2.1
```

Comment:
Out of scope for this PR - I've been playing with using `nix build .#ci.workspaceBuild` and extracting using zstd (referencing the logic in [inheritCargoArtifactsHook](https://github.com/ipetkov/crane/blob/7d5b46c17d857ee9ddb2e8d88185729a3e5637b6/lib/setupHooks/inheritCargoArtifactsHook.sh#L27)). That works well, but I can't get rustc to respect `FM_CARGO_DENY_COMPILATION`. Perhaps you know why?

I can open a separate PR to discuss.

Reply:
Also, back-compat is running fast enough where this is no longer the bottleneck in CI. Perhaps it's not worth the effort to solve the intermediate zstd extraction.

Reply:
> Perhaps you know why?

@bradleystachurski Need to make sure that timestamps, file ownership, location, etc. are all preserved. See [crane's version](https://github.com/ipetkov/crane/blob/2c94ff9a6fbeb9f3ea0107f28688edbe9c81deaa/lib/setupHooks/inheritCargoArtifactsHook.sh#L28)

> Perhaps it's not worth the effort to solve the intermediate zstd extraction.

Yeah, I think it's good enough right now.